jacoco-maven-plugin/src/org/jacoco/maven/AbstractReportMojo.java

```diff
@@ -186,12 +200,34 @@ private void executeReport(final Locale locale)
 		}
 	}
 
+	private void addFormatters(final ReportSupport support, final Locale locale)
+			throws IOException {
+		if (formats == null || formats.isEmpty()) {
+			support.addAllFormatters(getOutputDirectory(), outputEncoding,
+					footer, locale);
+		} else {
+			getOutputDirectory().mkdirs();
```

Comment:
Related to the above comment: How can we avoid this code duplication with ReportSupport methods? Maybe replace 

    addAllFormatters(...)

with 

    addFormatters(Set<Formats> ...)

Reply:
Idea: implement formatter creation in Enums.